include/libdnf/conf/option_binds.hpp

```diff
@@ -1,83 +1,91 @@
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_CONF_OPTION_BINDS_HPP
+#define LIBDNF_CONF_OPTION_BINDS_HPP
+
+
+#include "option.hpp"
 
 #include <functional>
 #include <map>
 
 namespace libdnf {
 
+/// Maps the options names (text names readed from config file, command line, ...) to options objects.
+/// Supports user defined functions for processing new value and converting value to string.
 class OptionBinds {
 public:
-    struct Exception : public std::runtime_error {
-        Exception(const std::string & what) : runtime_error(what) {}
-    protected:
-        mutable std::string tmpMsg;
+    struct Exception : public RuntimeError {
+        using RuntimeError::RuntimeError;
+        const char * get_domain_name() const noexcept override { return "libdnf::OptionBinds"; }
+        const char * get_name() const noexcept override { return "Exception"; }
+        const char * get_description() const noexcept override { return "OptionBinds exception"; }
     };
-    struct OutOfRange : public Exception {
-        OutOfRange(const std::string & id) : Exception(id) {}
-        const char * what() const noexcept override;
+
+    class OptionNotFound : public Exception {
+    public:
+        using Exception::Exception;
+        const char * get_name() const noexcept override { return "OptionNotFound"; }
+        const char * get_description() const noexcept override { return "Option not found"; }
     };
-    struct AlreadyExists : public Exception {
-        AlreadyExists(const std::string & id) : Exception(id) {}
-        const char * what() const noexcept override;
+
+    class OptionAlreadyExists : public Exception {
+    public:
+        using Exception::Exception;
+        const char * get_name() const noexcept override { return "OptionAlreadyExists"; }
+        const char * get_description() const noexcept override { return "Option already exists"; }
     };
 
+    /// Extends the option with user-defined functions for processing a new value and converting value to a string.
+    /// It is used as additional level of processing when the option is accesed by its text name.
     class Item final {
     public:
-        typedef std::function<void(Option::Priority, const std::string &)> NewStringFunc;
-        typedef std::function<const std::string & ()> GetValueStringFunc;
+        using NewStringFunc = std::function<void(Option::Priority, const std::string &)>;
+        using GetValueStringFunc = std::function<const std::string &()>;
```

Comment:
Why you change it? I just want to know why.

Reply:
@j-mracek It has the same meaning. From my point of view it is useless change.
But we set some rules and using `clang-tidy` for checking. And one  rule is to use `using` instead of `typedef`. So I changed it to have a less messages from `clang-tidy`.
If we agree, we can cancel this rule (https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-using.html).